common/djangoapps/course_modes/models.py

```diff
@@ -550,6 +550,32 @@ def is_professional_slug(cls, slug):
         """
         return slug in [cls.PROFESSIONAL, cls.NO_ID_PROFESSIONAL_MODE]
 
+    @classmethod
+    def has_masters_mode(cls, modes_dict):
+        """
+        Check whether the modes for a course contain a Master's mode.
```

Comment:
This doesn't really operate on a course, it operates on a dict

I'd probably use a word like "contains" rather than "has" in the name of this method to make more clear that this method works with less domain-focused data structures than a model